janusgraph-es/src/test/java/org/janusgraph/diskstorage/es/ElasticsearchIndexTest.java

```diff
@@ -323,6 +323,66 @@ public void testCustomMappingProperty() throws BackendException, IOException, Pa
         IOUtils.closeQuietly(response);
     }
 
+    @Test
+    public void testSetCardinality() throws Exception {
+        initialize("vertex");
+
+        Multimap<String, Object> initialDoc = HashMultimap.create();
+        initialDoc.put(PHONE_SET, "12345");
+
+        add("vertex", "test", initialDoc, true);
+
+        clopen();
+
+        Multimap<String, Object> updateDoc = HashMultimap.create();
+        updateDoc.put(PHONE_SET, "123456");
+
+        add("vertex", "test", updateDoc, false);
+
+        clopen();
+
+        add("vertex", "test", initialDoc, false);
+
+        clopen();
+
+        tx.delete("vertex", "test", PHONE_SET, "12345", false);
+
+        clopen();
+
+        assertEquals("test", tx.queryStream(new IndexQuery("vertex", PredicateCondition.of(PHONE_SET, Cmp.EQUAL, "123456"))).toArray()[0]);
+        assertEquals(0, tx.queryStream(new IndexQuery("vertex", PredicateCondition.of(PHONE_SET, Cmp.EQUAL, "12345"))).count());
+    }
+
+    @Test
+    public void testListCardinality() throws Exception {
```

Comment:
This method looks identical to `testSetCardinality` except the last line and `PHONE_SET`. Can move duplicate logic into a separate method? Something like:
```
public void testSetCardinality() throws Exception {
    testCollectionCardinality(PHONE_SET);
    assertEquals(0, tx.queryStream(new IndexQuery("vertex", PredicateCondition.of(PHONE_SET, Cmp.EQUAL, "12345"))).count());
}

public void testListCardinality() throws Exception {
    testCollectionCardinality(PHONE_LIST);
    assertEquals("test", tx.queryStream(new IndexQuery("vertex", PredicateCondition.of(PHONE_LIST, Cmp.EQUAL, "12345"))).toArray()[0]);
}

private void testCollectionCardinality(String collectionName) throws Exception{
    ...
}
```

Reply:
Done. But I think that with previous version was it was a little bit simpler to understand the idea of test, because it was easier to keep then «context» when all assertions were in single function.

Reply:
You can use parametrized test if you want to keep all association in the single function.
```
public static Stream<String> collectionNameParams() {
    return ImmutableList.of(PHONE_SET, PHONE_LIST).stream();
}

@ParametrizedTest
@MethodSource("collectionNameParams")
public void testCardinality(String collectionName){

   // you code test code...

  if(PHONE_SET.equals(collectionName)){
    assertEquals(0, tx.queryStream(new IndexQuery("vertex", PredicateCondition.of(PHONE_SET, Cmp.EQUAL, "12345"))).count());
  } else {
    assertEquals("test", tx.queryStream(new IndexQuery("vertex", PredicateCondition.of(PHONE_LIST, Cmp.EQUAL, "12345"))).toArray()[0]);
  }
}
```

Reply:
Good idea. Rewritten tests using this approach .